hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -39,7 +39,13 @@ import scala.collection.JavaConverters._
 import scala.collection.mutable
 import scala.util.Try
 
-case class HoodieMergeOnReadPartition(index: Int, split: HoodieMergeOnReadFileSplit) extends Partition
+/**
+ * Files a input task reads. Usually a file spilt corresponds to a task. For partitioned hudi table with bucket index,
```

Comment:
IIUC we just replace the first 8 of the fileID prefix right. so even though the bucket numbers are same, the file names will be different/unique across table?

Reply:
Yes. files with same bucketid but under the different partitions have different filenames